### DISCUSSION TITLE
in macOs,attempting to use xfreerdp for remote application in X11 environment resulted in failure

### DISCUSSION THREAD

**Discussion by YuanJieCao** [FreeRDP/FreeRDP#9802] [Q&A]
command is  ` xfreerdp  /u:test86 /p:test000  /v:192.168.6.43 /cert:ignore /app:program:chrome`.
in macOs, l donwloaded xfreerdp from homebrew.
 l saw your previous reply to someone where you mentioned macfreerdp doesn't work,is xfreerdp with +x11 okay?

**Comment by akallabeth** [comment]
* `xfreerdp` does support remote application mode
* native mac client `MacFreeRDP` and `sdl-freerdp` do currently not support it. see https://github.com/FreeRDP/FreeRDP/issues/9078
* your mileage with `xfreerdp` on mac-os may vary, it is limited by your `X11` environment (`XQartz`?)

**Comment by YuanJieCao** [comment]
l use XQartz. 
I use xfreerdp and can connect to the remote desktop normally. But when I used the remote app, it connected for the first time, displayed the desktop, and then displayed that the password was incorrect. The password was indeed incorrect, but after I entered the correct password, the command line displayed that the password was incorrect, and then the command exited.

**Reply by akallabeth** [reply]
@YuanJieCao doubt that, the `remote app` mode is only activated `AFTER` authentication, so if one mode succeeds to authenticate the other must as well.
maybe you can add some more details, aka log? (there are many error codes that are actually sent from the server and might mean something quite different than just `incorrect password`)

**Reply by YuanJieCao** [reply]
I compiled freerdp under macos and used it in the Xquartz environment, remote app. 
Yes, it can be used, but it cannot shrink and enlarge the window, and it will freeze if you click it.
Can you provide me with some ideas?
